robottelo/cli/factory.py

```diff
@@ -740,6 +741,53 @@ def make_role(options=None):
 
 
 @cacheable
+def make_filter(options=None):
+    """
+    Usage::
+
+        hammer filter create [OPTIONS]
+
+    Options::
+
+        --location-ids LOCATION_IDS         Comma separated list of values.
+        --locations LOCATION_NAMES          Comma separated list of values.
+        --organization-ids ORGANIZATION_IDS Comma separated list of values.
+        --organizations ORGANIZATION_NAMES  Comma separated list of values.
+        --permission-ids PERMISSION_IDS     Comma separated list of values.
+        --permissions PERMISSION_NAMES      Comma separated list of values.
+        --role ROLE_NAME                    User role name
+        --role-id ROLE_ID
+        --search SEARCH
+        -h, --help                          print help
+    """
+    args = {
+        u'location-ids': None,
+        u'locations': None,
+        u'organization-ids': None,
+        u'organizations': None,
+        u'permission-ids': None,
+        u'permissions': None,
+        u'role': None,
+        u'role-id': None,
+        u'search': None,
+    }
+
+    # Role and permissions are required fields.
+    if not options:
+        raise CLIFactoryError('Please provide required parameters')
+
+    # Do we have at least one role field?
+    if not any(options.get(key) for key in ['role', 'role-id']):
+        raise CLIFactoryError('Please provide a valid role field.')
+
+    # Do we have at least one permissions field?
+    if not any(options.get(key) for key in ['permissions', 'permission-ids']):
```

Comment:
So you have checks for role and permissions only. Just curious, is organization optional for filter or not? If it is, then corresponding check is needed too :)